### ISSUE TITLE
Can't install any apps on HyperOS/MIUI

### ISSUE THREAD

**Issue by ReduxFlakes** [ImranR98/Obtainium#1321]
**Describe the bug**
When trying to install any app trough obtainium, the same goes back to the home activity. When using the root method to install apps, Obtainium gives an error saying: Some errors occurred: [NewPipe]. When enabling the Google package installer on the system, obtainium simply crashes when clicking install on any app. I see a similar issue with Vendetta Manager where it also crashes and fails to install, however all of my other apps like any browser, Droidify, and so on, work correctly with any app installer like Installer X, App Manager, and the built-in one of HyperOS/MIUI. 

**To Reproduce**
1. Go to 'any app'
2. Tap on 'install'
4. See error of failing to install app, or the app crashing. 

**Screenshots and Logs**

https://github.com/ImranR98/Obtainium/assets/87225951/105366b6-6f98-4793-b04a-f352c520238d

![Screenshot_2024-01-21-13-37-51-722_dev imranr obtainium](https://github.com/ImranR98/Obtainium/assets/87225951/f3809278-6c6a-400b-afb6-a973e6a9766f)

---

 - Device: Xiaomi 11 Lite 5G NE
 - OS: HyperOS 1.0.24.1.8-DEV (also affects any MIUI China version) 
 - Obtainium Version: 0.15.11

**Comment by ImranR98** [comment]
MIUI is known to cause problems by "optimizing" Android APIs. Could you try and see if disabling optimizations fixes the issue? See here: https://github.com/ImranR98/Obtainium/issues/583#issuecomment-1763106004

**Comment by ReduxFlakes** [comment]
> MIUI is known to cause problems by "optimizing" Android APIs. Could you try and see if disabling optimizations fixes the issue? See here: [#583 (comment)](https://github.com/ImranR98/Obtainium/issues/583#issuecomment-1763106004)

Yes, disabling the option "Enable optimization" in HyperOS does make the Google package installer work, however that breaks half of the system, and some ROMs like Xiaomi.eu do remove that option because of the issues it causes, meaning Obtainium does not work at all on those ROMs. Thanks.

**Comment by ImranR98** [comment]
Reclassifying as en enhancement as it is closer to a new third-party install method like root or Shizuku. The normal Android API works.

**Comment by ImranR98** [comment]
I doubt I will be working on this (leaving the issue open for contributors). In the meantime, maybe the Shizuku installer would work for you?

**Comment by ReduxFlakes** [comment]
That would be great! I, unfortunately, can't test the Shizuku method since I'm back on crDroid which Obtainium works just fine, but with my experience with it, it should work.

**Comment by Gluneklairo** [comment]
I have this problems as well since several builds before, you can't use Root or Shizuku to install. By the way I recommend to add a download apk only option.

**Comment by BingoKingo** [comment]
Could you add **open an APK** instead of install it? Since a specific version update you have changed the installation API , it did not work anymore. It may resolve the issue for me on HyperOS.

**Comment by ImranR98** [comment]
This isn't a fix, but you can now download APKs separately into the Download folder, and open them with a file manager from there. See #1493.
